xtable-core/src/main/java/org/apache/xtable/catalog/ExternalCatalogConfigFactory.java

```diff
@@ -38,4 +46,16 @@ public static ExternalCatalogConfig fromCatalogType(
         .catalogProperties(properties)
         .build();
   }
+
+  private static <T> String findImplClassName(
```

Comment:
Should this just return the instance from line 52?

Reply:
I didn't quite get this how will we know which class to return without checking the catalogType ?

Reply:
Yes we can do that, similar to how we're already using the service loader

Reply:
Changed the signature for the function and renamed to `findInstance`.